internal/server/clear_task.go

```diff
@@ -0,0 +1,94 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+	"golang.org/x/sync/errgroup"
+	"net/http"
+	"strconv"
+)
+
+type ClearTaskClient interface {
+	ClearTask(ctx sirius.Context, taskID int) error
+	Task(ctx sirius.Context, id int) (sirius.Task, error)
+}
+
+type clearTaskData struct {
+	XSRFToken string
+	Entities  []string
+	Uid       string
+	Task      sirius.Task
+	Success   bool
+	Error     sirius.ValidationError
+}
+
+func ClearTask(client ClearTaskClient, tmpl template.Template) Handler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		if err := r.ParseForm(); err != nil {
+			return err
+		}
+
+		var lpa *sirius.Case
+		var taskIDs []int
+
+		for _, id := range r.Form["id"] {
+			taskID, err := strconv.Atoi(id)
+			if err != nil {
+				return err
+			}
+			taskIDs = append(taskIDs, taskID)
+		}
+		if len(taskIDs) == 0 {
+			return errors.New("no tasks selected")
+		}
+
+		ctx := getContext(r)
+		group, groupCtx := errgroup.WithContext(ctx.Context)
+
+		data := clearTaskData{XSRFToken: ctx.XSRFToken}
+		taskID, err := strconv.Atoi(r.FormValue("id"))
+		if err != nil {
+			return err
+		}
+
+		group.Go(func() error {
```

Comment:
This doesn't need to be inside a group, as it's a single request.

Do we only need to do this request if this is a GET?